Drop the  `.move` accessor

niksirbi opened the pull request:
## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [x] Other: removing an existing feature

**Why is this PR needed?**
Closes #319, for reasons explained in that issue.
Closes #257 as part of the course.

**What does this PR do?**
- [x] port functionalities we want to keep outside `move_accesor.py`
- [x] delete `move_accessor.py` outside
  - [x] some validations ported to `save_poses.py` (because they were only used upon saving anyway).
  - [x] class variables for dim_names and var_names moved to a new `movement_dataset.py` module, which now defines these per dataset type
- [x] Delete accessor tests, and modify the integration tests that used the accessor (they now use the functional syntax)
- [x] Remove all mentions of the accessor from the Getting Started guide in docs
- [x] Update examples to use the functional syntax instead of the accessor


> [!NOTE]
> This was less painful than I anticipated, thanks to our tests 🎉 
> I just had to remove the accessor class, see which test break.
> and remove or modify them accordingly.

## How has this PR been tested?

Yeah, by ensuring that existing test still pass, and full coverage is maintained.
Note that the overall coverage percentage has gone down by 0.01%, because although we still have the same number of uncovered lines, the total lines of code have gone down.

## Is this a breaking change?

**YES**: code that was using the accessor will break after the next release.

## Does this PR require an update to the documentation?

Yes, and it's been updated.

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

niksirbi commented:
Reminder:

Rebase to main after #295 is merged, to adopt the linkcheck config that was modified there.

lochhh reviewed (changes requested):
Thanks @niksirbi ! My comments are mostly related to validation. I think we don't need the `movement_dataset` module now, as all it does is defining constants - these can go into the respective dataset validators. I also added [cross-references](f7a4a4bb9dc3170f15b0ecc554580de75d08dec0) - the kinematics functions are lengthy but should be better once #328 and #320 are closed.

lochhh commented on movement/movement_dataset.py:
I think DIM_NAMES, VAR_NAMES can go into validators/datasets.py: ValidBboxesDataset, ValidPosesDataset, because this is part of our expectation of a valid Bboxes/Poses dataset. 
Not sure we need this module (yet?), since users are not expected to instantiate and use instances of MovementDataset, etc. - which may be confused with [the movement dataset](https://movement.neuroinformatics.dev/getting_started/movement_dataset.html) on the website. This module would make sense if we have something like BboxesDataset.load(...)

niksirbi replied:
You are right. We may need sth like this later, but not now.
Would you prefer to have `DIM_NAMES`, `VAR_NAMES` as class variables in the above validator classes, or on the `validators.py` module level? I guess the first makes mores sense?

lochhh replied:
Agreed. First one makes more sense.

niksirbi replied:
Alright, I implemented them as class variables in the respective dataset validator classes and removed the `movement_dataset.py` module.

Now that we have them there, I was tempted to fiddle with the validators to make the order of dimensions less hard-coded, but I stopped myself because that belong to #210 and #236.

niksirbi commented on movement/io/save_poses.py:
```diff
@@ -424,12 +425,25 @@ def _validate_dataset(ds: xr.Dataset) -> None:
-    ds.move.validate()  # validate the dataset
+
+    missing_vars = set(PosesDataset.get_var_names()) - set(ds.data_vars)
+    if missing_vars:
```
Agreed, I made a note under that issue.

niksirbi commented:
Thanks for the review @lochhh!. I've removed the `movement_dataset.py` module and moved `DIM_NAMES` and `VAR_NAMES` to class variable in their respective dataset validators. 

I also removed `codecov/project` as an absolute merge requirement, because as this PR shows, we don't care if the overall coverage % goes down (e.g. if we remove code). `codecov/patch` is still required (all new diff lines must be coverted).
